src/main/host/thread_ptrace.c

```diff
@@ -28,6 +28,12 @@
 
 #define THREADPTRACE_TYPE_ID 3024
 
+// glibc in centos 7 does not include the following, but it does use a supported
+// kernel (3.10 > 3.8)
+#if !defined(PTRACE_O_EXITKILL)
+#define PTRACE_O_EXITKILL (1 << 20)
+#endif
+
```

Comment:
Is this still needed without CentOS 7?

Reply:
Oops good catch. I ended up with a lot of merge conflicts after breaking out this PR to other PRs and merging them separately (each after some modification). I ended up just taking the final state of this PR and rebuilding on top of main from scratch, which meant I had to manually un-revert some changes. In hindsight this was probably not the best approach :)